robocop/checkers/naming.py

```diff
@@ -915,42 +907,50 @@ def visit_KeywordCall(self, node):  # noqa
 
     def visit_If(self, node):  # noqa
         for token in node.header.get_tokens(Token.ASSIGN):
-            self.check_for_reserved_naming(token, "Variable")
+            self.check_for_reserved_naming_or_hyphen(token, "Variable")
         self.generic_visit(node)
 
     def visit_Arguments(self, node):  # noqa
         for arg in node.get_tokens(Token.ARGUMENT):
-            self.check_for_reserved_naming(arg, "Argument")
+            self.check_for_reserved_naming_or_hyphen(arg, "Argument")
 
     def parse_embedded_arguments(self, name_token):
         """Store embedded arguments from keyword name. Ignore embedded variables patterns like (${var:pattern})."""
         try:
             for token in name_token.tokenize_variables():
                 if token.type == Token.VARIABLE:
-                    self.check_for_reserved_naming(token, "Embedded argument", has_pattern=True)
+                    self.check_for_reserved_naming_or_hyphen(token, "Embedded argument", has_pattern=True)
         except VariableError:
             pass
 
-    def check_for_reserved_naming(self, token, var_or_arg, has_pattern=False):
-        """Check if variable name is a reserved Robot Framework name."""
-        name, *_ = token.value.split("=", maxsplit=1)
-        name = name.rstrip()
+    def check_for_reserved_naming_or_hyphen(self, token, var_or_arg, has_pattern=False, is_assign=False):
```

Comment:
Not happy with this method but it should be fine for now. I can slightly refactor it in the (close) future)